packages/dds/tree/src/feature-libraries/indexing/anchorTreeIndex.ts

```diff
@@ -315,17 +302,18 @@ export class AnchorTreeIndex<TKey extends TreeIndexKey, TValue>
 	 * Checks if the spine needs to be re-indexed and if so, re-indexes it starting from the given path.
 	 */
 	private reIndexSpine(path: UpPath): void {
+		const cursor = this.forest.allocateCursor();
+		this.forest.moveCursorToPath(path, cursor);
+		assert(
+			cursor.mode === CursorLocationType.Nodes,
+			0xa9b /* attach should happen in a node */,
+		);
+		cursor.exitNode();
+		this.indexField(cursor);
```

Comment:
Was this a bug? If so, could we add a test for it?

Reply:
It's not clear to me whether it was or not. Before this change you may have been able to rely on the fact that indexed fields were updated through a replace, and if so, it was fine to only have this logic in replace. Now that replace is no longer used, I'm just moving this bit out of replace and into here so it gets run when needed. If you want to create a work item to investigate I'm happy to add a comment to refer to it.

Reply:
Oh actually I remember now that it's not a bug to have it, it's just unnecessary because in most cases, the afterCreate already handles it. So just moving to the isReplaced case is good.

Reply:
I don't think I'm following. What does "moving to the isReplaced case" mean precisely? Or perhaps it might be more straightforward to just tell me what the issue is with my changes.

Reply:
This logic that you moved into `reIndexSpine` should be instead be moved to the `isReplaced` branch of `afterDetach`. Otherwise, in the case of most attaches, this logic will be run twice, once in `afterCreate` and then again in `afterAttach`.

Reply:
Oh but I guess we do want to defer the indexing to the attach phase. Maybe lets leave this for now and just add a todo to optimize this later. We can use the same work item, I'll just change the title for it.

Reply:
That will lead to an assert because the object may be missing the node that is used to index it. This is what the comment in the `isReplaced` branch of `afterDetach` is talking about.

Reply:
Added the comment